OmniKitUI/ViewControllers/ReplacePodViewController.swift

```diff
@@ -15,37 +15,50 @@ class ReplacePodViewController: SetupTableViewController {
-                    break
+            case .fault(let eventType):
+                switch eventType {
+                case .reservoirEmpty:              // $18
+                    instructionsLabel.text = LocalizedString("Empty reservoir. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an empty reservoir")
+                case .exceededMaximumPodLife80Hrs: // $1C
+                    instructionsLabel.text = LocalizedString("Pod expired. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an expired pod")
+                case .occluded,                    // $14
+                     .occlusionCheckValueTooHigh, .occlusionCheckStartup1, .occlusionCheckStartup2,         // $5A, $60, $60
+                     .occlusionCheckTimeouts1, .occlusionCheckTimeouts2, .occlusionCheckTimeouts3,          // $62, $66, $67
+                     .occlusionCheckPulseIssue, .occlusionCheckBolusProblem, .occlusionCheckAboveThreshold: // $68, $69, $6A
+                    instructionsLabel.text = LocalizedString("Occlusion detected. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to an occlusion")
+                default:
+                    instructionsLabel.text = String(format: LocalizedString("The pod has detected internal fault 0x%02x. Insulin delivery has stopped. Please deactivate and remove pod.", comment: "Instructions when replacing pod due to a fault (1: The fault code value)"), eventType.rawValue)
                 }
-                
-                tableView.reloadData()
+            case .canceledPairingBeforeApplication:
+                instructionsLabel.text = LocalizedString("Incompletely setup pod must be deactivated before pairing with a new one. Please deactivate and discard pod.", comment: "Instructions when deactivating pod that has been paired, but not attached.")
+            case .canceledPairing:
+                instructionsLabel.text = LocalizedString("Incompletely setup pod must be deactivated before pairing with a new one. Please deactivate and remove pod.", comment: "Instructions when deactivating pod that has been paired and possibly attached.")
             }
+
+            tableView.reloadData()
         }
     }
     
     var pumpManager: OmnipodPumpManager! {
         didSet {
             pumpManager.getPodState { (podState) in
                 DispatchQueue.main.async {
-                    if podState?.fault != nil {
-                        self.replacementReason = .fault
+                    if let podFault = podState?.fault {
+                        if let faultType = podFault.currentStatus.faultType {
+                            self.replacementReason = .fault(faultType)
+                        } else {
+                            self.replacementReason = .fault(.valuesDoNotMatchOrAreGreaterThan0x97)
```

Comment:
Setting a .valuesDoNotMatchOrAreGreaterThan0x97 type when we don't have a faultType means that the UI will show 152 (0x98) for unknown fault codes, like if we happen to get an 0x04 back from the pod for some reason. If you have the .fault case store a PodInfoFaultEvent or FaultEventCode, then you'll have access to the actual fault codes for these.

Reply:
I thought that mapping unknown types to our code to a special fault code which indicates that the fault code didn't match anything would have been the preferred way to go on something like this, but I would have rather just printed the actual fault code. Sigh. Will try to rework again...

Reply:
This isn't a question of preference; it wasn't doing what you were wanting it to do (i.e. display the actual fault code).  It was displaying 0x98 for unknown codes that weren't 0x98.

Reply:
Yes, I understood this that would have been the case. In any case, no longer relevant with PodReplacementReason.fault now having the actual fault information.